src/main/java/fi/fmi/avi/archiver/message/populator/BulletinLocationIndicatorValidator.java

```diff
@@ -0,0 +1,57 @@
+package fi.fmi.avi.archiver.message.populator;
+
+import fi.fmi.avi.archiver.file.InputAviationMessage;
+import fi.fmi.avi.archiver.file.InputBulletinHeading;
+import fi.fmi.avi.archiver.message.ArchiveAviationMessage;
+import fi.fmi.avi.archiver.message.ProcessingResult;
+import fi.fmi.avi.model.GenericAviationWeatherMessage;
+import fi.fmi.avi.model.bulletin.BulletinHeading;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static fi.fmi.avi.archiver.message.populator.MessagePopulatorHelper.DEFAULT_BULLETIN_HEADING_SOURCES;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * When message aerodrome matches the given pattern, validate that the bulletin location indicator matches the given pattern.
+ */
+public class BulletinLocationIndicatorValidator implements MessagePopulator {
```

Comment:
I'd like the name of this class to describe that this validator grants a 'monopoly' to configured stations for configured senders. But I came up with no better suggestions so far. Any ideas?

Reply:
Inspired from suggestions for `MessageAerodromeValidator` below, how about `UnauthorizedOriginatorStationChecker`?